client/src/components/pub-search.jsx

```diff
@@ -0,0 +1,158 @@
+      navigate('/pub-search');
+    } else {
+      ApiService.get(query)
+        .then((response) => {
+          if (response.data === null) {
+            alert('No publication with that name found. Please enter it into the database.');
+            navigate('/pub-entry');
+          } else {
+            this.setState({
+              data: response.data,
+            });
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    }
+  };
+
+  addMyPubs = (event) => {
+    event.preventDefault();
+  };
+
+  render() {
+    if (this.state.data.length === 0) {
+      return (
+        <div>
+          <Link to="/pub-entry" className="btn btn-link text-secondary">
+            <span className="text-secondary">new publication</span>
+          </Link>
+          <h4>Search for Publication (by title or Twitter handle)</h4>
+          <div className="container-fluid">
+            <div className="container">
+              <div className="FormCenter">
+                <form className="FormField">
+                  <div className="FormField">
+                    <label for="searchTitle">Publication Title</label>
+                    <input
+                      type="text"
+                      className="form-control"
+                      id="searchTitle"
+                      placeholder="Enter name of publication"
+                      name="searchTitle"
+                      value={this.state.searchTitle}
+                      onChange={this.handleChange}
+                    />
+                  </div>
+                  <div className="form-group">
+                    <label for="searchTwitterHandle">Twitter Handle</label>
+                    <input
+                      type="text"
+                      className="form-control"
+                      id="searchTwitterHandle"
+                      placeholder="Enter Twitter Handle"
+                      name="searchTwitterHandle"
+                      value={this.state.searchTwitterHandle}
+                      onChange={this.handleChange}
+                    />
+                  </div>
+                  <br></br>
+                  <button className="btn btn-sm btn-secondary searchSubmit" onClick={this.handleSubmit}>
+                    Submit{' '}
+                  </button>
+                  <button className="btn btn-sm btn-secondary searchCancel" onClick={this.handleCancel}>
+                    Cancel{' '}
+                  </button>
+                </form>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <Link to="/pub-entry" className="btn btn-link text-secondary">
+            <span className="text-secondary">new publication</span>
+          </Link>
+          <Link to="/pub-search" className="btn btn-link text-secondary">
+            <span className="text-secondary">new search</span>
+          </Link>
```

Comment:
This doesn't work as expected because we're attempting to navigate to `/pub-search` from `/pub-search`. And there is no application refresh in the browser with Reach Router so it's like a do-nothing link click. In order to create a new search, we need to reset state.

Reply:
I found an inelegant fix. I tried setting the initial state as a variable but couldn't access it later when I needed it so simply reset the state to the initial values. But I'm sure there's a better way...